ang/civicase.ang.php

```diff
@@ -15,7 +15,16 @@
 use CRM_Civicase_Hook_Permissions_ExportCasesAndReports as ExportCasesAndReports;
 
 load_resources();
-$caseCategoryName = CRM_Utils_Request::retrieve('case_type_category', 'String');
+$caseCategoryId = CRM_Utils_Request::retrieve('case_type_category', 'Int');
+$caseCategoryName = '';
+if ($caseCategoryId > 0) {
+  $caseCategoryName = civicrm_api3('OptionValue', 'getsingle', [
+    'option_group_id' => 'case_type_categories',
+    'value' => $caseCategoryId,
+    'return' => ['name'],
+  ])['name'];
+}
+
 
```

Comment:
Don't we need to fetch the category ID with the default `case` category incase the case category is empty? Similar to what you did in the `ang/civicase-base.ang.php ` file? In that case we should probably extract that to a helper class and use the logic there and here.

Reply:
I doubted where to put this helper method, and I finally decided for `CaseUrl` helper. The CaseCategoryFromUrl sounds more appropriate, but it is actually returning the category name, not the category data (and in this case, is convenient to have both parameters calculated at once).
There were other options, but I think that CaseUrl is a good place, because by default we have the values of Cases category, which is what this class handles, in relation with URLs.
Done here: https://github.com/compucorp/uk.co.compucorp.civicase/pull/756/commits/d50dbe5f827b6d1d37f627d50f8f6008414e8322